### PR TITLE
replaced findbugs and checkstyle calls in Jenkinsfile.buildchain

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mareknovotny** [comment]
it seems ok to me, but we probably see the fix after merging only, is that correct?

**Comment by mbiarnes** [comment]
@mareknovotny @Ginxo @winklerm I run (replayed) this jobs here: [kie-soup-master.pr](https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-soup-master.pr/7/console)<br>It looked good. So I did the changes on [Jenkinsfile.buildchain](https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/Jenkinsfile.buildchain) and it seems it doesn't n like it too much.<br>What I am missing here?

**Comment by mbiarnes** [comment]
> it seems ok to me, but we probably see the fix after merging only, is that correct?
doesn't this PR take the upgraded Jenkinsfile.buildchain?

**Comment by mareknovotny** [comment]
@mbiarnes only problem could be in `12:28:35  WARNING: Unknown parameter(s) found for class type 'io.jenkins.plugins.analysis.core.steps.RecordIssuesStep': pattern` ? What is that unknown parameter(s) in https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-soup-master.pr/7/console?

**Comment by mbiarnes** [comment]
> @mbiarnes only problem could be in `12:28:35 WARNING: Unknown parameter(s) found for class type 'io.jenkins.plugins.analysis.core.steps.RecordIssuesStep': pattern` ? What is that unknown parameter(s) in https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-soup-master.pr/7/console?

@mareknovotny no idea - sorry

File: Jenkinsfile.buildchain [hunk 1]
```diff
@@ -143,13 +143,13 @@ pipeline {
             script {
                 if(findbugsFile) {
                     echo 'Findbugs reports ...'
-                    findbugs canComputeNew: false, defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', pattern: findbugsFile, unHealthy: ''
+                    recordIssues enabledForFailure: true, tool: spotBugs(), pattern: findbugsFile
```
Comment (mareknovotny) [inline Jenkinsfile.buildchain:146]: maybe you could extend the command from one line to more to be better readable ;)

File: Jenkinsfile.buildchain [hunk 2]
```diff
@@ -143,13 +143,13 @@ pipeline {
...
                 }
             }
             script {
                 if(checkstyleFile) {
                     echo 'Checkstyle reports ...'
-                    checkstyle canComputeNew: false, defaultEncoding: '', healthy: '', pattern: checkstyleFile, unHealthy: ''
+                    recordIssues enabledForFailure: true, tool: checkStyle(), pattern: checkstyleFile
```
Comment (mareknovotny) [inline Jenkinsfile.buildchain:152]: maybe you could extend the command from one line to more to be better readable ;)

File: Jenkinsfile.buildchain (hunk 1 above)
Comment (mareknovotny) [reply Jenkinsfile.buildchain:146]:
```suggestion
                    recordIssues (
                       enabledForFailure: true,
                       tool: findBugs(),
                       pattern: findbugsFile
                    )
```

File: Jenkinsfile.buildchain (hunk 2 above)
Comment (mareknovotny) [reply Jenkinsfile.buildchain:152]:
```suggestion
                    recordIssues(
                      enabledForFailure: true,
                      tool: checkStyle(), 
                      pattern: checkstyleFile
```

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**